kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ParquetColumnReaders.java

```diff
@@ -78,6 +78,8 @@ public static Converter createConverter(
                 initialBatchSize, (DecimalType) typeFromClient, typeFromFile);
         } else if (typeFromClient instanceof TimestampType) {
             return TimestampConverters.createTimestampConverter(initialBatchSize, typeFromFile);
+        } else if (typeFromClient instanceof VariantType) {
```

Comment:
Connectors that do not use `kernel-defaults` and do not require a knowledge of variant and just want kernel to extract fields could read this in as a struct... kernel can handle extracting